lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2321,6 +2322,16 @@ class HSADevice final : public KalmarDevice
 
 private:
 
+    void load_executables_once() {
+        if (!executables_loaded) {
```

Comment:
I think this is racy and probably needs to be protected by something like `std::call_once` ?